query-engine/driver-adapters/src/proxy.rs

```diff
@@ -173,169 +177,211 @@ pub struct Query {
     pub args: Vec<JSArg>,
 }
 
+fn conversion_error(args: &std::fmt::Arguments) -> QuaintError {
+    let msg = match args.as_str() {
+        Some(s) => Cow::Borrowed(s),
+        None => Cow::Owned(args.to_string()),
+    };
```

Comment:
Thanks to `ErrorKind::ConversionError` already expecting a `Cow<'_, str>`, this ensures `conversion_error!("some static string without interpolation")` is free, we only allocate when there are additional parameters.